colossalai/kernel/base_kernel_loader.py

```diff
@@ -0,0 +1,48 @@
+import platform
+from abc import ABC, abstractmethod
+from typing import Dict, List
+
+import torch
+
+from .extensions.base_extension import BaseExtension
+
+
+class BaseKernelLoader(ABC):
+    """
+    Usage:
+        kernel_loader = KernelLoader()
+        kernel = kernel_loader.load()
+    """
+
+    def __init__(self, extension_map: Dict[str, BaseExtension], supported_device: List[str]):
+        self._extension_map = extension_map
+        self._supported_device = supported_device
+
+    def run_checks(self):
+        # run supported device check and other possible checks
+        pass
+
+    @abstractmethod
+    def fetch_kernel(self):
+        pass
+
+    def load(self):
+        self.run_checks()
+        return self.fetch_kernel()
+
+    def _is_x86_available(self) -> bool:
+        return platform.processor() == "x86_64"
+
+    def _is_arm_available(self) -> bool:
+        return platform.processor() == "aarch64"
+
+    def _is_cuda_available(self) -> bool:
+        return torch.cuda.is_available()
+
+    def _is_npu_available(self) -> bool:
+        try:
+            import torch_npu  # noqa
+
+            return torch.npu.is_available()
+        except:
+            return False
```

Comment:
Such implementation is not extensible, if we are going to support other devices, we will need add more and more `_is_xxx_available` and use `if` and `else` to check what device we are running on.

Reply:
ok, how about just loop all the extensions and check these inside the extension?

Reply:
Would it be better to use `get_accelerator().is_available()`?

Reply:
when deciding what extension to build, we can use `get_accelerator().name`.

Reply:
okk